database-mongoose/reviews.service.js

```diff
@@ -162,13 +160,57 @@ let getReviewExcerpts = (productId) => {
 };
 
 let getSearchResults = (productId, searchString) => {
-  //to do
+
+  return ReviewsModel.
+
+    find({ productId: productId, description: { $regex: '.*' + searchString + '.*' } }).
+    limit(12).
+    sort({ isHelpfulCount: -1 }).
+    select({
+      reviewId: 1,
+      rating: 1,
+      productId: 1,
+      color: 1,
+      configuration: 1,
+      category: 1,
+      isBestSeller: 1,
+      customerId: 1,
+      customerName: 1,
+      customerCountry: 1,
+      title: 1,
+      description: 1,
+      isHelpfulCount: 1,
+      isVerifiedPurchase: 1,
+      imageUrls: 1,
+      profilePicUrl: 1,
+      reviewDate: 1
+    })
+    .then(results => {
+      console.log(results);
+      var resultsPromise = new Promise((resolve, reject) => {
+        resolve(results)
+      })
+      return resultsPromise;
+
+    })
+    .catch(err => console.log(err))
 
 }
 
+let incrementHelpfulCount = (reviewId) => {
+  return ReviewsModel.updateOne({ reviewId: reviewId }, { $inc: { 'isHelpfulCount': 1 } })
+    .then(results => {
+      var resultsPromise = new Promise((resolve, reject) => { resolve(results) })
+      return resultsPromise;
+    })
+    .catch(err => console.log(err))
+
+}
 
```

Comment:
New database functions look good. Consider using `const` for these function expressions.